### ISSUE TITLE
Text/Icon contrast correction on dashboard and sidebar

### ISSUE THREAD

**Issue by matthewpereira** [concourse/concourse#5974]
## Summary

Update key components to ensure that we meet [WCAG accessibility guidelines for color contrast](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast). This will ensure that differently sighted users or users interacting with the web UI on poor displays or in poor lighting conditions can still operate comfortably and accurately, and the overall interface will have consistent visual cues across multiple elements and views. In addition, having a planned source of truth for our colours means less guess work and technical debt when adding or modifying the Elm codebase. 

The goal of this issue is to move from this current 'before' state:

![Before](https://user-images.githubusercontent.com/6372810/90661353-19026780-e215-11ea-81d5-9992ab09dbd6.png)

To this 'after' state:

![After](https://user-images.githubusercontent.com/6372810/90779372-c1730300-e2cb-11ea-8380-8087b3306bac.png)

While the changes are subtle at a glance, they ensure that every icon and label is more readable, and set a baseline for consistency that we can extend to the rest of the app. 

![After with Labels](https://user-images.githubusercontent.com/6372810/90778993-51fd1380-e2cb-11ea-8a46-17890f353bfc.png)

The color codes relate to a [scale that's available for quick reference in Figma](https://www.figma.com/file/fuoFSVcTGptiS21P2POA2XbN/Concourse-UI-kit-and-Experiments?node-id=1608%3A19867):

![Color Scale](https://user-images.githubusercontent.com/6372810/90779330-b0c28d00-e2cb-11ea-91af-3cb5f6af7aa9.png)

## Acceptance Criteria:

Match the following elements to the 'After' states above:
- [ ] Background in CI monitor, sidebar, header bar, legend bar, pipeline tiles
- [ ] Text labels on pipeline tiles
- [ ] Icons on pipeline tiles
- [ ] Disabled state for icons or buttons
- [ ] Sidebar icons and text
- [ ] Header and legend bar icons and text labels
- [ ] Tooltip text
- [ ] Search box

## Not included in this story:
X Don't alter key colours (success/failure/error/abort/pinned/paused) yet, as we'll want to update these across other views as well.

## Todo:
- [ ] Illustrate hover states for sidebar

## Context

* Part of #5964 Web UI/UX Experience

**Comment by scottietremendous** [comment]
Let's add some acceptance criteria before this is picked up.

**Comment by jomsie** [comment]
So just doing a quick look-see.

Most (like 99%) of the colour definitions are in the `web/elm/src/Colors.elm`
There are three files (`web/elm/src/Views/Styles.elm`, `web/elm/src/SideBar/Styles.elm`, `web/elm/src/Dashboard/Styles.elm`) that have single hex colour strings within them.

So this _might_ be a `small` instead of a `medium`. (Not completely sure yet, but if so, @taylorsilva I beg forgiveness.)

**Comment by jomsie** [comment]
I'm going to post my research and findings here. (Need to keep my thoughts straight with the Colors.elm file).

Across the app, a single variable in the Colors.elm file can be used many times for many different reasons (see Colors.frame below). Changing just the dashboard/sidebar (as this issue intends), we'll have to be careful of what we change.

Current strategy would be to keep the current variables for everything not affected by this story, but introduce new (for the time being) variables, for any changes. Perhaps add a bit more granularity for the possibility of theming (👀) down the road.

**Colors.frame**
Used:
- Build Styles (7x)
- **Dashboard Styles** (2x)
- Login Styles (2x)
- Pipeline Styles (2x)
- PinMenu Styles (3x)
- Resource Styles (1x)
- **Sidebar/Sidebar.elm** (2x)
- **Sidebar Styles** (4x)
- **Views Styles** (1x)
(in bold = relevant to this issue)

Note: tooltips will go #363636 text, #E2E2E2 bg

**Comment by jomsie** [comment]
Questions for matt:
1. on the new HD page, we have a little orange triangle top right when there's an error (i think broken resource), which colour should that be?
2. can i get exact values for the HD page (like success40 for success etc)

**Comment by matthewpereira** [comment]
1. I think it should be `Error 40` - #DE951D
2. [Check this frame in Figma for the colour codes](https://www.figma.com/file/fuoFSVcTGptiS21P2POA2XbN/Concourse-UI-kit-and-Experiments?node-id=1602%3A4)
3. Search drop down background can just be `Grey 90` - #262626. Do you mean highlighted/non highlighted search results? Could you share a screenshot of what you mean?
4. Can we try Sidebar colors as follows and see how they look?
Default: text `Grey 30` #C6C6C6, background `Grey 90` #262626
Hover: text `Grey 20` #E2E2E2, background `Grey 80` #363636
Active: text `Grey 20` #E2E2E2, background `Grey 100` #151515
5. I think that's fine. WDYT?

**Comment by jomsie** [comment]
THANKS FOR EVERYTHING!

ps. I'm cool with the 5

**Comment by jomsie** [comment]
found some less files that have some good ol hex codes in there. continuing!

** base00 ** -> #1e1d1d <img width="16" alt="Screen Shot 2020-09-25 at 2 59 53 PM" src="https://user-images.githubusercontent.com/2362482/94305790-c6b51480-ff3f-11ea-8791-acd77c1e001e.png">
 
used: 
css/colors.less 
  - for: builds, builds list, build step header, topbar, build number
    - probably only care about topbar? but i can't find a top-bar id anywhere, so i'll skip
css/login.less
  - for: login-box
    - probably could change this
css/modules.less
  - for: list-collapsable-item, jobs-build-list/resource, 
css/svg.less
  - for: node-rect (pipeline drawin')

**Comment by jomsie** [comment]
further updates:

doing a search for `"#` throughout the code base to try to find the white/black/grey ones.

**Comment by jomsie** [comment]
further further updates:

sidebar has a bunch of colour changes based on opacity, luckily 
1. team icon / team name won't change colour
2. the +/-'s also won't change colour

just the pipeline name, icon, and fav icon will change depending

**Comment by jomsie** [comment]
Update on this one.

Got the pipeline icon / unfilled fav icon working with default and a bright versions of the svg 
Pulled out more opacity code
Fixing all the tests

Still have to do:
rename opacity to something brightness/colour related for the sidebar stuff
fix the top bar bg's on the pipeline view, the little buttons stop right 
merge in #5969

**Comment by chenbh** [comment]
Just noticed that if you are part of multiple teams and have the sidebar open, selecting a pipeline will make the team name brighter: `rgb(226, 226, 226)` -> `rgb(255, 255, 255)`. But the change is so small I had to squint really hard just to make sure it's there.

![Screen Shot 2020-12-03 at 4 07 49 PM](https://user-images.githubusercontent.com/17577073/101088511-d2493b00-3581-11eb-9c44-978c7fa48320.png)
![Screen Shot 2020-12-03 at 4 07 56 PM](https://user-images.githubusercontent.com/17577073/101088514-d2e1d180-3581-11eb-9a1b-03fd702c18fc.png)
